packages/trackerless-network/src/logic/StreamEntryPointDiscovery.ts

```diff
@@ -69,6 +70,7 @@ export class StreamEntryPointDiscovery {
     private readonly config: StreamEntryPointDiscoveryConfig
     private readonly servicedStreamParts: Map<StreamPartID, NodeJS.Timeout>
     private readonly cacheInterval: number
+    private readonly networkSplitAvoidedNodes: Map<StreamPartID, Set<NodeID>> = new Map()
```

Comment:
Is entry point discovery for each stream a separate task? I.e. to discover an entry point for stream part S1, do we need any info we possible gathered previously when entry points for stream part S2 was discovered?

If that's the case, we could make `StreamEntryPointDiscovery` to operate just with an explicit stream part and create a separate instance each time we create a new `StreamObject` in `StreamrNode`. The `knownStreamEntryPoints` could be also a field in `StreamObject`, or embedded to the `StreamEntryPointDiscovery` class?

Reply:
Yes, the class could be refactored to be per stream. However this might make future features like caching previously found stream entry points more difficult. This could be useful in cases where a node leaves and rejoins a stream in a quick succession.